src/handler/promote.ts

```diff
@@ -0,0 +1,97 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import 'dotenv/config';
+import { cloneDeep } from 'lodash';
+import { TechRecordGet } from '../models/post';
+import { PromoteRecordRequestBody } from '../models/promote';
+import { SearchCriteria } from '../models/search';
+import { archiveOldCreateCurrentRecord, getBySystemNumberAndCreatedTimestamp, searchByCriteria } from '../services/database';
+import { getUserDetails } from '../services/user';
+import { addHttpHeaders } from '../util/httpHeaders';
+import logger from '../util/logger';
+import { validatePromoteErrors } from '../validators/promote';
+
+export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayProxyResult> => {
+  logger.info('Promote end point called');
+
+  const promoteErrors = validatePromoteErrors(event);
+
+  if (promoteErrors) {
+    return addHttpHeaders(promoteErrors);
+  }
+
+  const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber as string ?? '');
+  const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp as string ?? '');
```

Comment:
```suggestion
  const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp ?? '');
```
just out of curiosity, does this not work?

Reply:
I tried this on the archive endpoint. The linter definitely does not like this at all and insisted on the "as string" approach.